apps/src/labs/projects/ProjectManager.ts

```diff
@@ -29,7 +29,9 @@ export default class ProjectManager {
   private saveFailListeners: ((response: Response) => void)[] = [];
   private saveStartListeners: (() => void)[] = [];
   private lastSource: string | undefined;
-  private channel: Channel | undefined;
+  private sourceToSave: Source | undefined;
+  private lastChannel: Channel | undefined;
```

Comment:
It might be worth a comment on this variable, and possibly others too?

Reply:
added!